azurerm/internal/services/containers/kubernetes_nodepool.go

```diff
@@ -281,6 +284,10 @@ func ExpandDefaultNodePool(d *schema.ResourceData) (*[]containerservice.ManagedC
 		profile.ProximityPlacementGroupID = utils.String(proximityPlacementGroupId)
 	}
 
+	if upgradeSettingsRaw, ok := raw["upgrade_settings"].([]interface{}); ok && len(upgradeSettingsRaw) > 0 {
```

Comment:
we need to expand this regardless (since that'll allow us to revert this to the defaults too):

```suggestion
	profile.UpgradeSettings = expandUpgradeSettings(raw["upgrade_settings"].([]interface{})
```